### ISSUE TITLE
Firelocks

### ISSUE THREAD

**Issue by D3athrow-Issues** [d3athrow/vgstation13#14275]
(WEB REPORT BY: kyunny REMOTE: 172.93.109.202:7777)
Can someone make firelocks deconstrucable?
they fuck my my room rearrangement projects.
>Revision (Should be above if you're viewing this from ingame!)

>General description of the issue

>What you expected to happen

>What actually happened

>Steps to reproduce if possible

**Comment by abagone** [comment]
I agree there are a couple things that really need some construction/deconstruction code like the Access Consoles(little panels that let you cycle airlocks) cannot be deconstructed.

**Comment by abagone** [comment]
Like I find it stupid how you can build a fire alarm but you cannot build the fire locks or deconstruct them. 

I think it would be something simple like some metal, wires, and a atmospheric sensor to build them. 

**The steps to build could be.**

1 Create fire lock frame 

2 use a wrench to secure it to the floor 

3 Add wires to the assembly 

4 Insert atmospheric sensor 

5 secure with screwdriver 

**And to deconstruct.**

1 Screwdriver it open 

2 crowbar the atmospheric sensor out 

3 remove wires 

4 unwrench it from the floor 

5 wrench the fire lock frame to deconstruct it

**Comment by Kurfursten** [comment]
code it

**Comment by Exxion** [comment]
Aren't they already deconstructable? You just need the socket wrench.

**Comment by Probe1** [comment]
They're deconstructable with socket wrenches.

**Comment by abagone** [comment]
The **socket wrench** why? 
Is there any good reason to make it difficult to deconstruct Fire locks?
I bet it's like this because someone slapped on a way to deconstruct them. 
There should really be a proper way to build and dismantle them. Like what I posted above.

**Comment by Exxion** [comment]
No, it was specifically made that way so that it would be difficult to do. Don't want some dunderfuck going around and putting a hole in every firelock.

Honestly, though, just taking a really long time to deconstruct would be sufficient to prevent that.

**Comment by abagone** [comment]
A firelock is just a mini airlock that has no access and reacts to atmospheric changes.

I feel that firelocks should be the same/easier to deconstruct as a airlock and not require the socket wrench. Maybe in the steps above I could include welding them as well?

It should not be this **almost unmovable object** that **cannot be rebuilt!**

> Exxion
No, it was specifically made that way so that it would be difficult to do. Don't want some dunderfuck going around and putting a hole in every firelock.

Also why shouldn't people be able to mess with the firelocks they can mess with just about **everything else?** Like for example deconstruct the **walls around the firelocks** rendering them **useless** and the **wall is 10X times easier to remove than the firelock!**

The socket wrenche is uncommon for a reason because it can detach pipes that are full.
We cannot make the socket wrench more common to make removing firelocks easier so it should not be tied to the firelocks.

TLDR: Firelocks need to be build able and not tied to the socket wrenche because they're too difficult to get and should not be made more common because they break pipes easily.

**Comment by Exxion** [comment]
I don't think anyone's actually disagreeing with you here
All I said was that there *was* thought behind making it require the socket wrench, regardless of whether or not that thought was good

**Comment by Kurfursten** [comment]
nobody's arguing with you dude

if you put as much time in coding as you did preaching to the choir, it would already be merged right now

**Comment by abagone** [comment]
Actually since my last comment I have been trying to learn BYOND code.
After that I just need to figure out how Github works.

**Comment by SonixApache** [comment]
firelocks are the heroes the station needs

**Comment by PJB3005** [comment]
I was doubting your claim that embedded controllers can't be deconstructed but you're right, mostly.

Some oldcoder twat decided that mapped in embedded are invincible. Should be a simple enough fix.

**Comment by kernikov** [comment]
Firelocks cannot be deconstructed, welded or unwelded, open or shut, by socket wrenches.

**Comment by Exxion** [comment]
Huh. That was certainly an idea, which I could have sworn got implemented, but apparently did not.

Also, on a mostly unrelated note, I found this fucking gem in firelock code while looking at it to see if you were right:
```
/obj/machinery/door/firedoor/attack_hand(mob/user as mob)
	return attackby(null, user)
```
And then the following special case in `attackby()` so that that actually works (because it would just fuck everything up ruinously otherwise):
```
if(!istype(C, /obj)) //If someone hit it with their hand.  We need to see if they are allowed.
	if(allowed(user))
		access_granted = 1
	if(ishuman(user))
		users_name = FindNameFromID(user)
	else
		users_name = "Unknown"
```

I mean... I can sort of understand why they did it (so everything about checking access and recording who opened it could be handled in `attackby()`), but there are so many better alternatives
Also, of note is the fact that this is one of the few cases in which `isobj()` is actually preferable to `istype(/obj)` and yet they still used the latter
(Yes, those should be equivalent. But they aren't. Thanks, Lummox.)

OK sorry to hijack the issue but seeing `if(!istype(C, /obj))` in `attackby()` was really surreal for a moment before I saw the `attack_hand()` part

**Comment by PJB3005** [comment]
Firelocks were made deconstruct able by Kurf but then comic changed them to directional ones and didn't readd it.

**Comment by mtpowell** [comment]
Since you're all talking about firelocks, on Box(not sure if it exists on Def/Meta) the firelocks are OUTSIDE of the tile of the door in the main hallways. 

Everyother door has them on the same tile, thus I can open the door/firelock in two clicks. If they're on separate tiles I need to open one, wait for it to open than open the other.

They're on the same tile literally in every other god damn place on any other map.

**Comment by kernikov** [comment]
Box is the worst for firelocks, imo.

**Comment by ihadtoregisterforthis** [comment]
Fulfilled
